### PR TITLE
Diversity survey 2021 updates

### PR DESCRIPTION
**Pull request by jamescodeorg**
Update and enable the diversity survey for 2021.  The survey has been simplified this year to have just one question and no follow-up survey.

I need to get a final ok from Amanda on layout, colors, and copy on Monday but this is a bit time-sensitive so hoping to get it reviewed a bit early.

Screen Shot 2021-03-29 at 5 01 44 PM

Screen Shot 2021-03-29 at 5 21 37 PM

## Links

- spec: [K-5 diversity survey 2021](https://docs.google.com/document/d/1ISv5MHsBlIuNND5lHwdnSV5OUqa7LlJBcTbCJ61yR_A/edit#)
- jira ticket: [LP-1800](https://codedotorg.atlassian.net/browse/LP-1800)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/survey_result.rb
Comment (bencodeorg) [inline dashboard/app/models/survey_result.rb:49]: Is there any reason for serialized attributes to still include keys used in older responses? I don't think we read from this model anywhere, so it probably doesn't matter.

**Review by bencodeorg — APPROVED** [review]
LGTM pending style input from Amanda (I dunno about that green...)

File: dashboard/app/views/home/_survey_diversity.haml
Comment (bencodeorg) [inline dashboard/app/views/home/_survey_diversity.haml]: I feel like there should be a colon here, if you're in there making style updates before you merge.

File: dashboard/app/models/survey_result.rb
Comment (jamescodeorg) [reply dashboard/app/models/survey_result.rb:49]: You know what?  I don't think this constant is used anywhere.  I'm just going to delete the whole list and see what happens.  Thanks!

File: dashboard/app/views/home/_survey_diversity.haml
Comment (jamescodeorg) [reply dashboard/app/views/home/_survey_diversity.haml]: I will double check this with Amanda (along with the green, which I agree looks a bit suspect).

File: dashboard/app/models/survey_result.rb
Comment (bencodeorg) [reply dashboard/app/models/survey_result.rb:49]: Interesting, I was thinking serialized_attrs was a Rails validation thing that determined what serialized attributes could be saved. Sounds like you already figured this out but apparently it's just [homebaked](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/concerns/serialized_properties.rb#L50) and gives you a list of serialized properties, like `SurveyResult.serialized_attrs`? Also [here's some unrelated helpful variable naming](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/concerns/serialized_properties.rb#L42). 😆

Comment (jamescodeorg) [reply dashboard/app/models/survey_result.rb:49]: Heh, I often see `klass` used as a variable name when referring to a class, not sure if I've seen `clazz` before.  (You can't call the variable `class` because it's a reserved keyword.)

I took another look and realized that I misunderstood how the survey values were getting into the database.  I had thought that the JSON was just being sanitized and then saved to the db, but it's actually being parsed and then each value is being stored as a serialized attribute on the model.  Thanks for pointing this out!  (Side note: This design couples the db model to the survey questions which seems odd to me.)

To answer your original question, I'll leave the old values in the list with a note that they're not used anymore in case anyone wants to access the old values via the model.

File: dashboard/app/models/survey_result.rb
Comment (bencodeorg) [inline dashboard/app/models/survey_result.rb:37]: Nit: other is deprecated as well, yeah?

**Review by bencodeorg — APPROVED** [review]
Latest commit LGTM -- if you have a moment to update the screenshot in the PR description (looks like the success message is updated, but the dialog is not?), that'd be nice.

File: dashboard/app/models/survey_result.rb
Comment (jamescodeorg) [reply dashboard/app/models/survey_result.rb:37]: Oh right, thanks for pointing that out.  I'm going to go ahead and merge but will update if I'm back in here again.

**Pull request merged**
